File: LeetCode/DataStruct/Python/Leetcode901-online-stock-span.py
class StockSpanner:

    def __init__(self):
        self.idx = 0
        self.stk = [[-1, int(1e6)]]

    def next(self, price: int) -> int:
        while price >= self.stk[-1][1]:
            self.stk.pop()
        res = self.idx - self.stk[-1][0]
        self.stk.append([self.idx, price])
        self.idx += 1
        return res

File: LeetCode/DataStruct/JS/Leetcode901-online-stock-span.js
var StockSpanner = function() {
    this.stk = [[-1, parseInt(1e6)]];
    this.idx = 0;
};

/** 
 * @param {number} price
 * @return {number}
 */
StockSpanner.prototype.next = function(price) {
    while(price >= this.stk[this.stk.length - 1][1]) {
        this.stk.pop();
    }
    let res = this.idx - this.stk[this.stk.length - 1][0];
    this.stk.push([this.idx, price]);
    this.idx += 1;
    return res;
};
